benchmarks/api_benchmark.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Microbenchmarks for JAX `api` functions."""
+import functools
+import operator
+
+import jax
+import jax.numpy as jnp
+
+import benchmark
+
+
+def required_devices(num_devices_required):
+  """Helper to skip benchmarks that require more devices."""
+  def helper1(f):
+    @functools.wraps(f)
+    def helper2(state):
+      if jax.device_count() < num_devices_required:
+        state.skip_with_error(f"requires {num_devices_required} devices")
+        return
+      return f(state)
+    return helper2
+  return helper1
+
+
+@benchmark.register
+def jit_trivial(state):
+  f = jax.jit(swap)
+  a, b = f(1, 2)
+
+  while state:
+    a, b = f(a, b)
```

Comment:
Using assignment seems like a poor practice here, since it breaks the independent assumption of each loop?

Also, I think you probably want to use `block_until_ready()`, e.g.,
```suggestion
    f(a, b).block_until_ready()
```

Reply:
I think the goal here is to measure dispatch time and IIRC `jit(swap)` does not actually result in an xlacomputation so  the block_until_ready is not necessary:

```python
c, d = jit(lambda a, b: (b, a))(a, b)
assert c is b and d is a
```

Agree we can probably skip assignment.